apps/designer/app/designer/features/style-panel/shared/style-info.ts

```diff
@@ -0,0 +1,38 @@
+import type { Style, StyleProperty, StyleValue } from "@webstudio-is/css-data";
+import { useMemo } from "react";
+
+export type StylePropertyInfo = {
+  source: "local" | "external";
+  value: StyleValue;
+};
+
+export type StyleInfo = {
+  [property in StyleProperty]?: StylePropertyInfo;
+};
+
+export const useStyleInfo = ({
```

Comment:
I think filename must me use-style-info

Reply:
1. locks with single hook export
2. we often provide non-hook api

I prefer to name files as types rather than utilities of these types

Reply:
I call them by the function name if its the only exported function

Reply:
this helps preventing adding more functions to it, but sometimes it makes sense to put more functions into the same module, in which case `style-info` kind of naming makes more sense